apps/web/src/studio/components/workflows/step-editor/WorkflowTestStepButton.tsx

```diff
@@ -19,10 +21,16 @@ export const WorkflowTestStepButton = ({
   workflowId: string;
   stepType: ChannelTypeEnum;
 }) => {
+  const segment = useSegment();
+  const isStudio = useIsStudio();
   const { currentUser } = useAuth();
   const { mutateAsync: testSendEmailEvent, isLoading: isTestingEmail } = useMutation(testSendEmailMessage);
 
   const handleTestClick = async () => {
+    segment.track('Step test ran', {
+      step: ChannelTypeEnum.EMAIL,
+      env: isStudio ? 'local' : 'cloud',
+    });
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** maybe add the context, like Email step test run - [Workflows Test Page]‏